internal/trace/context.go

```diff
@@ -126,21 +127,36 @@ func createDummySubsegmentForXrayConverter(ctx context.Context, traceCtx TraceCo
 	return nil
 }
 
-func getTraceContext(context map[string]string) (TraceContext, bool) {
+func getTraceContext(ctx context.Context, headers map[string]string) (TraceContext, bool) {
```

Comment:
When the extension is in charge of parsing for trace context, it is included in the response from calling start invocation. It is then added to the context object. If present, it therefore should be read in case there is no trace context found in the event payload headers.

Reply:
by event payload headers, you mean the actual event that triggered the go lambda right?
Just wanna make sure I understand right, so we get trace context by either pinging the extension [here](https://github.com/DataDog/datadog-agent/blob/570ed026d8dc3afc4324610db8f922027f8786ca/pkg/serverless/daemon/routes.go#L60) which populates `ctx`

and `headers` could have trace context via some event such as api gw like [this](https://github.com/DataDog/datadog-agent/blob/570ed026d8dc3afc4324610db8f922027f8786ca/pkg/serverless/trace/testdata/event_samples/http-api.json#L15)?

So to be sure we extract and attach context we do check both ways.

Reply:
That is exactly correct. Reading the "event payload headers" is hold over from before we added universal instrumentation. We could potentially get rid of it completely and just rely on the extension to do this work for us.